lib/shopify-cli/js_system.rb

```diff
@@ -0,0 +1,95 @@
+require 'shopify_cli'
+
+module ShopifyCli
+  ##
+  # ShopifyCli::JsSystem allows conditional system calls of npm or yarn commands.
+  #
+  class JsSystem
+    include SmartProperties
+
+    YARN_CORE_COMMAND = 'yarn'
+    NPM_CORE_COMMAND = 'npm'
+
+    class << self
+      ##
+      # Proxy to instance method ShopifyCli::JsSystem.new.yarn?.
+      #
+      # #### Parameters
+      # - `ctx`: running context from your command
+      #
+      # #### Example
+      #
+      #   ShopifyCli::JsSystem.yarn?(ctx)
+      #
+      def yarn?(ctx)
+        JsSystem.new(ctx: ctx).yarn?
+      end
+
+      ##
+      # Proxy to instance method ShopifyCli::JsSystem.new.call.
```

Comment:
being pedantic bc it's OSS but should the  `ShopifyCli::JsSystem.new.call` be in backticks so that it's clear the ending period belongs to the sentence and is not part of the call

Reply:
I'm not sure but in other examples when explaining it is proxied it was not in backticks. I decided to follow the previous convention.  Example here: https://github.com/Shopify/shopify-app-cli/blob/03e251363bbdcb7a23c7ded671e5d47c657a5fcc/lib/shopify-cli/js_deps.rb#L13